core/src/main/java/de/jplag/highlight_extraction/FrequencyDetermination.java

```diff
@@ -0,0 +1,77 @@
+package de.jplag.highlight_extraction;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import de.jplag.*;
+
+/**
+ * Calculates frequencies of match subsequences across all comparisons according to different strategies.
+ */
+
+public class FrequencyDetermination {
+    private Map<String, List<String>> tokenFrequencyMap = new HashMap<>();
+    private FrequencyStrategy freqencyStrategy;
+    private int strategyNumber;
+
+    /**
+     * Constructor.
+     * @param frequencyStrategy The chosen strategy for frequency calculation.
+     * @param strategyNumber Parameter used by certain strategies to determine submatch length.
+     */
+    public FrequencyDetermination(FrequencyStrategy frequencyStrategy, int strategyNumber) {
+        this.freqencyStrategy = frequencyStrategy;
+        this.strategyNumber = strategyNumber;
+    }
+
+    /**
+     * Applies the "create" and "check" methods of the frequency strategy.
+     * @param comparisons contains information of matches between two submissions.
+     * @return Map containing (sub-)matches and their frequency according to the strategy.
+     */
+    public Map<String, List<String>> runAnalysis(List<JPlagComparison> comparisons) {
+        frequencyBuilder(comparisons, freqencyStrategy::create);
+        frequencyBuilder(comparisons, freqencyStrategy::check);
```

Comment:
I am not the biggest fan of the functional interface here. Either provide the strategy as an instance, or make this class abstract, merge the `FrequencyStrategy` into this class, and make the strategies subclasses (https://en.wikipedia.org/wiki/Strategy_pattern).